server/api.go

```diff
@@ -45,12 +50,58 @@ type PRDetails struct {
 	Reviews            []*github.PullRequestReview `json:"reviews"`
 }
 
+type HTTPHandlerFuncWithUser func(w http.ResponseWriter, r *http.Request, userID string) http.HandlerFunc
+
 func writeAPIError(w http.ResponseWriter, err *APIErrorResponse) {
 	b, _ := json.Marshal(err)
 	w.WriteHeader(err.StatusCode)
 	w.Write(b)
 }
 
+func (p *Plugin) initialiseAPI() {
+	p.router = mux.NewRouter()
+
+	webHookRouter := p.router.PathPrefix(API_WEBHOOK).Subrouter()
+	oauthRouter := p.router.PathPrefix(API_OAUTH).Subrouter()
+	apiRouter := p.router.PathPrefix(API_PREFIX).Subrouter()
```

Comment:
I'm unsure if it's really needed to have the routes be defined as global `const`. Is this needed for testing?